File: web/characters.js
const CHARACTERS = [
  { id: 'scorpion', name: 'Alakran', emoji: '🦂' },
  { id: 'liukang', name: 'Fueguito', emoji: '🔥' },
  { id: 'johnnycage', name: 'Galancito', emoji: '🕶️' },
  { id: 'kunglao', name: 'Sombrerito', emoji: '🎩' },
  { id: 'kitana', name: 'Abaniquita', emoji: '🪭' },
  { id: 'mileena', name: 'Colmillita', emoji: '🦷' },
  { id: 'reptile', name: 'Lagartijo', emoji: '🦎' },
  { id: 'subzero', name: 'Hielito', emoji: '🧊' },
  { id: 'jax', name: 'Bracitos', emoji: '💪' },
  { id: 'baraka', name: 'Navajita', emoji: '🗡️' },
  { id: 'shangtsung', name: 'Abuelito', emoji: '🧙' },
  { id: 'smoke', name: 'Humito', emoji: '💨' },
];

function characterByID(id) {
  return CHARACTERS.find((c) => c.id === id) || CHARACTERS[0];
}

function saveCharacter(id) {
  try { localStorage.setItem('kxp-character', id); } catch (e) {}
}

function loadCharacter() {
  try { return localStorage.getItem('kxp-character') || CHARACTERS[0].id; } catch (e) { return CHARACTERS[0].id; }
}

File: characters.go
package main

import "math/rand/v2"

type Character struct {
	ID   string
	Name string
}

var characters = []Character{
	{ID: "scorpion", Name: "Alakran"},
	{ID: "liukang", Name: "Fueguito"},
	{ID: "johnnycage", Name: "Galancito"},
	{ID: "kunglao", Name: "Sombrerito"},
	{ID: "kitana", Name: "Abaniquita"},
	{ID: "mileena", Name: "Colmillita"},
	{ID: "reptile", Name: "Lagartijo"},
	{ID: "subzero", Name: "Hielito"},
	{ID: "jax", Name: "Bracitos"},
	{ID: "baraka", Name: "Navajita"},
	{ID: "shangtsung", Name: "Abuelito"},
	{ID: "smoke", Name: "Humito"},
}

func characterByID(id string) (Character, bool) {
	for _, c := range characters {
		if c.ID == id {
			return c, true
		}
	}
	return Character{}, false
}

func defaultCharacterID() string {
	return characters[0].ID
}

func validCharacter(id string) bool {
	_, ok := characterByID(id)
	return ok
}

func randomCharacterID() string {
	return characters[rand.IntN(len(characters))].ID
}
